src/useCookieConsent.ts

```diff
@@ -13,21 +13,33 @@ import { allCookiesSetToValue, allPropsApproved } from './utils';
 
 export const useCookieConsent = (
   options?: CookieConsentOptions
-): CookieConsentHookState => {
+): ConsentHookStateType => {
```

Comment:
If you describe the type of the function instead of defining just it's return type then you wouldn't need any generics or conditional Types to connect arguments to the return type, you would just have two types for the function instead of two types for the options in the arguments and two for return types.